src/workerd/io/io-context.c++

```diff
@@ -1082,7 +1082,47 @@ void IoContext::runImpl(Runnable& runnable, bool takePendingEvent,
       js.terminateExecution();
     }
 
-    js.runMicrotasks();
+    // Running the microtask queue can itself trigger a pending exception in the isolate.
+    v8::TryCatch tryCatch(workerLock.getIsolate());
+
+    try {
+      js.runMicrotasks();
+    } catch (jsg::JsExceptionThrown&) {
```

Comment:
I think this can't possibly catch anything, since `js.runMicrotasks()` directly calls into V8 through which exceptions don't pass.

I guess it's interesting that `PerformMicrotaskCheckpoint()` doesn't return a value, therefore doesn't return any indication that a termination exception could have been thrown, yet `TryCatch` will catch it.

Reply:
Yeah, I put this here purely defensively but there's likely no need.